src/io/io.c

```diff
@@ -440,8 +440,11 @@ char* MVM_platform_path(MVMThreadContext *tc, MVMString *path) {
     char *original_path = MVM_string_utf8_c8_encode_C_string(tc, path);
 
 #ifdef _WIN32
-    // Check if the path already starts with "\\?\"
-    if (is_absolute_path(original_path) == 1 && strncmp(original_path, "\\\\?\\", 4) != 0) {
+    // Add prefix if:
+    // * It is an absolute path
+    // * It doesn't already start with \\?\
+    // * It is at least MAX_PATH in length
+    if (is_absolute_path(original_path) == 1 && strlen(original_path) > MAX_PATH && strncmp(original_path, "\\\\?\\", 4) != 0) {
```

Comment:
I suspect this `strlen(original_path) > MAX_PATH` will get CI passing. I think more generally something could check if the path contains a `.` or `..` and either not apply the prefix or to somehow resolve the path to not contain those relative path parts.

Reply:
Indeed it seems to have worked.